File: migrate.py
import os

with open('/tmp/pip_css.css', 'r') as f:
    css = f.read()

with open('/tmp/pip_html.html', 'r') as f:
    html = f.read()

with open('NewOrigami.Panels.html', 'r') as f:
    panels = f.read()

# Insert CSS into Panels
panels = panels.replace('    </style>', css + '\n    </style>', 1)

# Insert HTML into #main-container
panels = panels.replace('<div id="main-container">', '<div id="main-container">\n' + html, 1)

with open('NewOrigami.Panels.html', 'w') as f:
    f.write(panels)

with open('NewOrigami.FPV.1.html', 'r') as f:
    fpv = f.read()

fpv = fpv.replace(css, '')
fpv = fpv.replace(html, '')

with open('NewOrigami.FPV.1.html', 'w') as f:
    f.write(fpv)

File: migrate.js
const fs = require('fs');

const css = fs.readFileSync('/tmp/pip_css.css', 'utf-8');
const html = fs.readFileSync('/tmp/pip_html.html', 'utf-8');

let panels = fs.readFileSync('NewOrigami.Panels.html', 'utf-8');

// Insert CSS right before </style>
panels = panels.replace('    </style>', css + '\n    </style>');

// Insert HTML into #main-container
panels = panels.replace('<div id="main-container">', '<div id="main-container">\n' + html);

fs.writeFileSync('NewOrigami.Panels.html', panels);

// Clean up FPV.1.html
let fpv = fs.readFileSync('NewOrigami.FPV.1.html', 'utf-8');
// Remove the CSS block (lines 263-468)
fpv = fpv.replace(css, '');
// Remove the HTML block (lines 500-520)
fpv = fpv.replace(html, '');

fs.writeFileSync('NewOrigami.FPV.1.html', fpv);

// We still need to migrate initWindowDragging logic!
